Moved the logic that checks if content_type is text/event-stream

danechitoaie opened the pull request:
Moved the logic that checks for content_type into a separated function _check_content_type that is now called from iter_sse and aiter_sse instead of the constructor.

This should allow now to check if request was successful or not and handle the errors.

See #11 

This should allow now for:
```python
import httpx
from httpx_sse import connect_sse

with httpx.Client() as client:
    with connect_sse(client, "GET", "http://localhost:8000/sse") as event_source:
        if not event_source.response.is_success:
            # handle errors
            pass

        for sse in event_source.iter_sse():
            print(sse.event, sse.data, sse.id, sse.retry)
```

florimondmanca reviewed (commented):
Makes perfect sense, and we’ll executed, thanks!

Could we add a test to `test_event_source.py` for this, or update any previous test that checked the behavior when an unexpected content type was received? I believe that’s the one causing failure on CI; it needs updating.

danechitoaie commented:
I've updated test_connect_sse_non_event_stream_received with attempting to iterate over `event_source.iter_sse()` as that's where we moved the check and that's where the exception is raised now.

florimondmanca reviewed (approved):
Super, thanks!
